app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,27 @@
+class SessionsController < ApplicationController
+
+  def new
+    @user = User.new
+  end
+
+  def login
+    @user = User.find_by(name: params[:user][:name])
+
+    if @user
+      session[:user_id] = @user.id
+      flash[:success] = "#{ @user.name } is successfully logged in"
+    else
+      @user = User.create(name: params[:user][:name])
+      flash[:success] = "New user #{ @user.name } successfully created"
+    end
```

Comment:
You create a new user here but don't add their user ID to the session, so they won't actually be logged in. This means that a new user has to enter their username twice.